src/directives/NgGrid.ts

```diff
@@ -1072,11 +1072,11 @@ export class NgGrid implements OnInit, DoCheck, OnDestroy {
 								}
 							}
 
-							if (lowest != itemPos.col) {	//	If the item is not already on this col move it up
+							if (lowest != itemPos.col && this._isWithinBoundsX({lowest, r}, itemDims))) {	//	If the item is not already on this col move it up

```

Comment:
That is correct. This only happened with GridItems that are larger than the standard (1 row wide) size. I also added the same rule for up/down cascading.